src/Terminal/Run.gren

```diff
@@ -45,91 +51,159 @@ type alias MakeConfig msg =
 type alias RunConfig msg =
     { cpPermission : ChildProcess.Permission
     , fsPermission : FileSystem.Permission
-    , path : String
+    , pathToString : Path -> String
+    , path : Path
     , onInit : Process.Id -> msg
     , onExit : Int -> msg
     }
 
 
+type alias ProjectOutline =
+    { projectPath : Path
+    , outline : Outline
+    }
+
+
 make : MakeConfig msg -> Task Error (Cmd msg)
 make config =
-    let
-        installPackages nullStream projectOutline =
-            PackageInstall.run
-                { fsPermission = config.fsPermission
-                , cpPermission = config.cpPermission
-                , useColor = config.useColor
-                , interactive = False
-                , stdout = nullStream |> Stream.writable
-                , stdin = nullStream |> Stream.readable
-                }
-                projectOutline
-                    |> Task.mapError PackageInstallError
-
-        backendCommand outputType outputPath packageResolution =
+    when config.package is
+        Just packageName ->
+            clonePackage config packageName
+                |> Task.andThen (makeProject config)
+        Nothing ->
+            PackageInstall.readProjectOutline config.fsPermission
+                |> Task.mapError ReadProjectOutlineError
+                |> Task.andThen (makeProject config)
+
+
+makeProject : MakeConfig msg -> ProjectOutline -> Task Error (Cmd msg)
+makeProject config projectOutline =
+    when getOutputPath projectOutline is
+        Nothing ->
+            Task.fail NotAnApplication
```

Comment:
Do we want this restriction? 🤔 

Makes it more hassle to test the programs "shipped" with `gren run --package`

Reply:
Unless I'm missing something this would only happen if `platform=common`, which means you can't have browser or node packages which means you can't run any modules in your package.